### ISSUE TITLE
[Bug]: Xcode 15.3: Failed to build module 'OneSignalFramework'; this SDK is not supported by the compiler

### ISSUE THREAD

**Issue by rursache** [OneSignal/OneSignal-iOS-SDK#1380]
### What happened?

The SDK does not compile on Xcode 15.3

### Steps to reproduce?

```Markdown
1. Add onesignal to any project via SPM
2. Use Xcode 15.3 and run the project
```

### What did you expect to happen?

Project to compile

### OneSignal iOS SDK version

5.1.1

### iOS version

15

### Specific iOS version

```Markdown
15.4
```

### Relevant log output

Failed to build module 'OneSignalFramework'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)', while this compiler is 'Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)'). Please select a toolchain which matches the SDK.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by coskungun** [comment]
I have this problem too. I started getting this error after upgrading to the latest version of Xcode.


Failed to build module 'OneSignalFramework'; this SDK is not supported by the compiler (SDK is built with 'Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)', this compiler is 'Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)'). Please select a toolchain that matches the SDK.

**Comment by masahirocom** [comment]
I am running into the same problem.
I also have the following error in the framework
'OSInAppMessages' is not a member type of class 'OneSignalCore.

**Comment by rursache** [comment]
thanks @coskungun, indeed using version 5.1.0 of onesignal sdk works with xcode 15.3

**Comment by nan-li** [comment]
Hi, thanks for reporting, we had an issue with this release, we shipped an updated version in [Release 5.1.3](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/5.1.3).

**Comment by masahirocom** [comment]
> Hi, thanks for reporting, we had an issue with this release, we shipped an updated version in [Release 5.1.3](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/5.1.3).
Could you please update the following?
OneSignal-XCFramework(https://github.com/OneSignal/OneSignal-XCFramework)
